### PR TITLE
Created an export command for pattern

### LINKED ISSUE #10195: Support OCI export of Designs, Patterns, Filters
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

Presently Meshery UI supports exporting as OCI (Open Container Initiative), Using mesheryctl, export in OCI format for enhancing sharing and portability of design.

This is child issue of #6447

### Desired Behavior

Mesheryctl should support exporting Designs, Patterns, Filters in OCI format

### Acceptance Tests

- [ ] mesheryctl should support the export of Designs, Patterns, Filters
- [ ] Update documentation

### Mockups

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Althaf66**
**Notes for Reviewers**

This PR fixes #10195 

Created `mesheryctl pattern export` command for exporting design to OCI format. Now mesheryctl support exporting the design.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: mesheryctl/internal/cli/root/pattern/export.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+			utils.Log.Error(err)
+			return nil
+		}
+		pattern := ""
+		isID := false
+
+		if len(args) > 0 {
+			pattern, isID, err = utils.ValidId(mctlCfg.GetBaseMesheryURL(), args[0], "pattern")
+			if err != nil {
+				utils.Log.Error(ErrPatternInvalidNameOrID(err))
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/pattern/export.go:71]: Reminder: we will change from the word “pattern” to the word “design” very soon.

File: mesheryctl/internal/cli/root/pattern/export.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+				return nil
+			}
+		}
+
+		urlString := mctlCfg.GetBaseMesheryURL()
+		if isID {
+			// if pattern is a valid uuid, then directly fetch the pattern
+			urlString += "/api/pattern/" + pattern
+		} else {
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/pattern/export.go:80]: Trim the argument. 

What else should we be checking for here? Be sure to be practicing secure coding.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: mesheryctl/internal/cli/root/pattern/export.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			utils.Log.Error(err)
+			return nil
+		}
+		pattern := ""
+		isID := false
+
+		if len(args) > 0 {
```
[the remark this answers is not in the document]
Comment (Althaf66) [reply mesheryctl/internal/cli/root/pattern/export.go:68]: What are the other things as argument?

File: mesheryctl/internal/cli/root/pattern/export.go (hunk 1 above)
Comment (Althaf66) [reply mesheryctl/internal/cli/root/pattern/export.go:71]: I will change to design

File: mesheryctl/internal/cli/root/pattern/export.go (hunk 2 above)
Comment (Althaf66) [reply mesheryctl/internal/cli/root/pattern/export.go:80]: We checking the id is valid or not ,if it is valid, then searching the pattern by its id else it searching for the pattern by its name. there is different api for id and name.

File: mesheryctl/internal/cli/root/pattern/export.go (hunk 1 above)
Comment (leecalcote) [reply mesheryctl/internal/cli/root/pattern/export.go:71]: @Althaf66 good, although, only after the `app` to `pattern` transition is complete and is bug-free.

**Comment by leecalcote** [comment]
@RipulHandoo @alphaX86 your review here would be great.

**Comment by leecalcote** [comment]
Unit tests failing, @Althaf66

**Comment by Althaf66** [comment]
Yes, I am working on that

**Comment by Yashsharma1911** [comment]
@RipulHandoo can we merge this PR?

**Comment by leecalcote** [comment]
@humblenginr FYI

**Comment by leecalcote** [comment]
Pull request abandoned by author.

**Pull request closed without merging**